pegen/grammar_visualizer.py

```diff
@@ -1,82 +1,114 @@
 import argparse
 import tokenize
+import functools
 import enum
 
 from pegen.tokenizer import Tokenizer
 from pegen.tokenizer import grammar_tokenizer
 from pegen.grammar2 import GrammarParser
 
-
-argparser = argparse.ArgumentParser(prog='pegen', description="Experimental PEG-like parser generator")
+argparser = argparse.ArgumentParser(prog='pegen', description="Pretty print the AST for a given PEG grammar")
 argparser.add_argument('filename', help="Grammar description")
 
 
 class Direction(enum.Enum):
-        UP = 0
-        DOWN = 1
-        NEUTRAL = 2
-        NONE = 3
-
-
-class GrammarPrinter:
-
-    def children(self, node):
+    UP = 0
+    DOWN = 1
+    NEUTRAL = 2
+    NONE = 3
+
+
+class ASTGrammarPrinter:
+    _direction_to_start_shape = {
+        Direction.UP: '┌',
+        Direction.DOWN: '└',
+        Direction.NEUTRAL: ' ',
+        Direction.NONE: '├'
+    }
+
+    @staticmethod
```

Comment:
Just because something doesn't use self, why make it a static method? That decorator is meant for things you can call directly on the class, but here `GrammarPrinter.children(node)` doesn't make sense.